controllers/mars/pod.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The Alibaba Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/alibaba/kubedl/api/marsjob/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/alibaba/kubedl/pkg/job_controller"
+	"github.com/alibaba/kubedl/pkg/util"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *MarsJobReconciler) GetPodsForJob(obj interface{}) ([]*corev1.Pod, error) {
```

Comment:
similarly, this file also contains duplication for other job types, maybe open an issue and fix later